integration/pull_test.go

```diff
@@ -845,3 +853,42 @@ func TestRpullImageThenRemove(t *testing.T) {
 
 	checkFuseMounts(t, sh, 0)
 }
+
+// TestRpullImageWithMinLayerSize pulls and rpulls an image with a runtime min_layer_size to confirm small layers are mounted locally
+func TestRpullImageWithMinLayerSize(t *testing.T) {
+	regConfig := newRegistryConfig()
+	sh, done := newShellWithRegistry(t, regConfig)
+	defer done()
+
+	// Setup environment
+	if err := testutil.WriteFileContents(sh, defaultContainerdConfigPath, getContainerdConfigYaml(t, false), 0600); err != nil {
+		t.Fatalf("failed to write %v: %v", defaultContainerdConfigPath, err)
+	}
+	if err := testutil.WriteFileContents(sh, defaultSnapshotterConfigPath, getSnapshotterConfigYaml(t, false, minLayerSizeConfig), 0600); err != nil {
+		t.Fatalf("failed to write %v: %v", defaultSnapshotterConfigPath, err)
+	}
+	sh.
+		X("update-ca-certificates").
+		Retry(100, "nerdctl", "login", "-u", regConfig.user, "-p", regConfig.pass, regConfig.host)
+
+	rebootContainerd(t, sh, "", "")
+
+	containerImage := rabbitmqImage
+
+	copyImage(sh, dockerhub(containerImage), regConfig.mirror(containerImage))
+	indexDigest := buildIndex(sh, regConfig.mirror(containerImage))
```

Comment:
We should probably pass `WithMinLayerSize(0)` to ensure that ztocs for all layers are being built.

Reply:
That feels like a different case that might deserve its own test. What I'd like to be doing here is comparing to the default behavior.

Reply:
https://github.com/awslabs/soci-snapshotter/pull/363 just changed the default behavior for tests from 0 to `defaultMinLayerSize`. `WithMinLayerSize(0)` is probably equivalent to what you actually tested against.

Reply:
Good catch! That would explain the discrepancy.

Reply:
I believe that I've addressed this.